service/src/main/java/bio/terra/externalcreds/services/ProviderService.java

```diff
@@ -164,6 +154,47 @@ public void deleteLink(String userId, String providerId) {
     linkedAccountService.deleteLinkedAccount(userId, providerId);
   }
 
+  @VisibleForTesting
+  void authAndRefreshPassport(LinkedAccount linkedAccount) {
+    if (linkedAccount.getExpires().before(Timestamp.from(Instant.now()))) {
+      passportService.deletePassport(linkedAccount.getId().get());
+      linkedAccountService.updateLinkAuthenticationStatus(linkedAccount.getId().get(), false);
+    } else {
+      try {
+        var clientRegistration =
+            providerClientCache.getProviderClient(linkedAccount.getProviderId());
+        var accessTokenResponse =
+            oAuth2Service.authorizeWithRefreshToken(
+                clientRegistration.orElseThrow(),
```

Comment:
That's a good point! If we change this we'd want to do it across the code base, but since it's not specific to this PR I think we'll want to discuss as a team and decide so I'm going to leave it for now.

Reply:
Sounds good! Just updated it.